llvm/lib/Transforms/Utils/Local.cpp

```diff
@@ -1029,11 +1031,12 @@ CanRedirectPredsOfEmptyBBToSucc(BasicBlock *BB, BasicBlock *Succ,
     return false;
 
   // Get single common predecessors of both BB and Succ
+  // TODO: Replace this with a proper `set intersect` algorithm
   for (BasicBlock *SuccPred : SuccPreds) {
     if (BBPreds.count(SuccPred)) {
+      CommonPred = SuccPred;
```

Comment:
This change makes no sense -- did you run the tests for this PR at all?

This is finding a single common predecessor, so it bails out if there are multiple. After your change it bails out if there are any common predecessors.

Reply:
yeah, this is not needed. The second bug is somewhere else in simplifycfg, i'm still investigating.